db/dataset_eval.py

```diff
@@ -0,0 +1,129 @@
+from db import create_cursor, commit
+import db.dataset
+import db.data
+import jsonschema
+
+# Job statuses are defined in `eval_job_status` type. See schema definition.
+STATUS_PENDING = "pending"
+STATUS_RUNNING = "running"
+STATUS_DONE = "done"
+STATUS_FAILED = "failed"
+
+
+def evaluate_dataset(dataset_id):
+    """Add dataset into evaluation queue.
+
+    Args:
+        dataset_id: ID of the dataset that needs to be added into the list of
+            jobs.
+
+    Returns:
+        ID of the newly created evaluation job.
+    """
+    with create_cursor() as cursor:
+        cursor.execute(
+            "SELECT count(*) FROM dataset_eval_jobs WHERE dataset_id = %s AND status IN %s",
+            (dataset_id, (STATUS_PENDING, STATUS_RUNNING))
+        )
+        if cursor.fetchone()[0] > 0:
+            raise JobExistsException
+    validate_dataset(db.dataset.get(dataset_id))
+    return _create_job(dataset_id)
+
+
+def validate_dataset(dataset):
+    """Validate dataset by making sure that it matches JSON Schema for complete
+    datasets (JSON_SCHEMA_COMPLETE) and checking if all recordings referenced
+    in classes have low-level information in the database.
```

Comment:
For now, this exception is OK. But could we do something more automatically? Perhaps we just ignore mbids with no match in AB. But maybe we need a flag for the user to say it's OK.